src/MetadataUtility/Metadata/SupportFiles/FLLogFile.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="FLLogFile.cs" company="QutEcoacoustics">
+// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group.
+// </copyright>
+
+namespace MetadataUtility.Metadata.SupportFiles
+{
+    using System.Text.RegularExpressions;
+    using LanguageExt;
+
+    public static class FLLogFile
+    {
+        public const string FrontierLabsLogString = " FRONTIER LABS Bioacoustic Audio Recorder ";
+
+        public static Fin<bool> FileExists(TargetInformation information)
+        {
+            List<string> logFiles = new List<string>();
+
+            // If support file directories were given, search there for log files, if not search the directory of the audio file
+            if (information.SupportFileDirectories == null)
+            {
+                logFiles = information.FileSystem.Directory.GetFiles(information.FileSystem.Path.GetDirectoryName(information.Path), "*logfile*.txt", SearchOption.AllDirectories).ToList();
+            }
+            else
+            {
+                foreach (string dir in information.SupportFileDirectories)
+                {
+                    logFiles.AddRange(information.FileSystem.Directory.GetFiles(dir, "*logfile*.txt", SearchOption.AllDirectories));
+                }
+            }
+
+            foreach (string logFile in logFiles)
+            {
+                string[] lines = information.FileSystem.File.ReadAllLines(logFile);
+
+                if (lines[2].Equals(FrontierLabsLogString))
```

Comment:
I would scan the first few lines for the header - all sorts of weird behaviour could cause this header to be off by a line